test/jdk/java/util/UUID/UUIDTest.java

```diff
@@ -228,32 +295,39 @@ private static void nodeTest() throws Exception {
         } catch (UnsupportedOperationException uoe) {
             // Correct result
         }
+
         test = UUID.fromString("00000001-0000-1000-8001-000000000001");
-        if (test.node() != 1)
+        if (test.node() != 1) {
             throw new Exception("Incorrect node");
+        }
+
         test = UUID.fromString("00000001-0000-1000-8002-FFFFFFFFFFFF");
-        if (test.node() != ((2L<<47)-1)) // 2^48 - 1
+        if (test.node() != ((1L << 48) - 1)) {
             throw new Exception("Incorrect node");
+        }
     }
 
     private static void hashCodeEqualsTest() throws Exception {
         // If two UUIDs are equal they must have the same hashCode
-        for (int i=0; i<100; i++) {
+        for (int i = 0; i < COUNT; i++) {
             UUID u1 = UUID.randomUUID();
             UUID u2 = UUID.fromString(u1.toString());
-            if (u1.hashCode() != u2.hashCode())
-                throw new Exception("Equal UUIDs with different hashcodes");
+            if (u1.hashCode() != u2.hashCode()) {
+                throw new Exception("Equal UUIDs with different hash codes: " + u1 + " and " + u2);
```

Comment:
Perhaps we should even print the hash codes that weren't matching to provide assistance when debugging?

Reply:
Yes, added.